tests/test_models/test_reservation_unit_pricing_updates.py

```diff
@@ -12,64 +15,263 @@
+
+    future_pricing_count = ReservationUnitPricing.objects.filter(status=PricingStatus.PRICING_STATUS_FUTURE).count()
+    assert future_pricing_count == 1  # New pricing should be created for the change date
+
+    future_pricing = ReservationUnitPricing.objects.filter(status=PricingStatus.PRICING_STATUS_FUTURE).first()
+    assert future_pricing.tax_percentage.value == FUTURE_TAX
+    assert active_pricing.tax_percentage.value == CURRENT_TAX  # Active pricing should not be changed
+
+
+def test_reservation_unit__update_pricings__tax_percentage__future_pricing_before_change_date():
+    pricing_1 = ReservationUnitPricingFactory.create(
+        begins=datetime.date(2024, 1, 1),
+        status=PricingStatus.PRICING_STATUS_ACTIVE,
+        tax_percentage__value=CURRENT_TAX,
+        highest_price=Decimal("10"),
+    )
+    pricing_2 = ReservationUnitPricingFactory.create(
+        begins=datetime.date(2024, 8, 31),
+        status=PricingStatus.PRICING_STATUS_FUTURE,
+        reservation_unit=pricing_1.reservation_unit,
+        tax_percentage__value=CURRENT_TAX,
+        highest_price=Decimal("20"),
+    )
+
+    update_reservation_unit_pricings_tax_percentage(str(TAX_CHANGE_DATE), str(CURRENT_TAX), str(FUTURE_TAX))
+
+    future_pricing_count = ReservationUnitPricing.objects.filter(status=PricingStatus.PRICING_STATUS_FUTURE).count()
+    assert future_pricing_count == 2  # New pricing should be created, since the future pricing is before change date
+
+    future_pricing = ReservationUnitPricing.objects.filter(
+        status=PricingStatus.PRICING_STATUS_FUTURE,
+        begins=TAX_CHANGE_DATE,
+    ).first()
+    assert future_pricing.tax_percentage.value == FUTURE_TAX
+    # Price should be the same as in the last pricing before the change date
+    assert future_pricing.highest_price == pricing_2.highest_price
+    assert future_pricing.highest_price_net < pricing_2.highest_price_net
+
+
+@pytest.mark.parametrize(
+    "latest_begins_date",
+    [
+        TAX_CHANGE_DATE,
+        datetime.date(2024, 9, 21),  # After the change date
+    ],
+)
+def test_reservation_unit__update_pricings__tax_percentage__future_pricing_after_change_date(latest_begins_date):
+    pricing_1 = ReservationUnitPricingFactory.create(
+        begins=datetime.date(2024, 1, 1),
+        status=PricingStatus.PRICING_STATUS_ACTIVE,
+        tax_percentage__value=CURRENT_TAX,
+    )
+    ReservationUnitPricingFactory.create(
+        begins=latest_begins_date,
+        status=PricingStatus.PRICING_STATUS_FUTURE,
+        reservation_unit=pricing_1.reservation_unit,
+        tax_percentage__value=CURRENT_TAX,
+    )
+
+    update_reservation_unit_pricings_tax_percentage(str(TAX_CHANGE_DATE), str(CURRENT_TAX), str(FUTURE_TAX))
+
+    future_pricing_count = ReservationUnitPricing.objects.filter(status=PricingStatus.PRICING_STATUS_FUTURE).count()
+    assert future_pricing_count == 1  # No new pricings should be created, since a pricing after the change date exists
```

Comment:
Should test that we get a log to sentry about pricing that should be handled manually.